internal/controller/tokenratelimitpolicy_status_updater.go

```diff
@@ -0,0 +1,256 @@
+				overriddenByMessage = fmt.Sprintf(" by %s", overriddenBy[0])
+			}
+			overriddenPolicyRules = append(overriddenPolicyRules, fmt.Sprintf("%s%s", policyRuleKey, overriddenByMessage))
+		}
+		message = fmt.Sprintf("policy rule(s) overridden: %s", strings.Join(overriddenPolicyRules, ", "))
+	}
+
+	if len(affectedGateways) == 0 { // no rules of the policy found in the effective policies
+		if len(overridingPolicies) == 0 { // no rules of the policy have been overridden by any other policy
+			return kuadrant.EnforcedCondition(policy, kuadrant.NewErrNoRoutes(policyKind), false)
+		}
+		// all rules of the policy have been overridden by at least one other policy
+		overridingPoliciesKeys := lo.FilterMap(lo.Uniq(lo.Flatten(lo.Values(overridingPolicies))), func(policyLocator string, _ int) (k8stypes.NamespacedName, bool) {
+			policyKey, err := kuadrantpolicymachinery.NamespacedNameFromLocator(policyLocator)
+			return policyKey, err == nil
+		})
+		return kuadrant.EnforcedCondition(policy, kuadrant.NewErrOverridden(policyKind, overridingPoliciesKeys), false)
+	}
+
+	// check if any policy rule of the policy has been overridden by a more specific policy
+	if len(overridingPolicies) > 0 {
+		if message != "" {
+			return kuadrant.EnforcedCondition(policy, kuadrant.NewErrUnknown(policyKind, fmt.Errorf("%s", message)), false)
+		}
+		// all rules of the policy have been overridden by at least one other policy
+		overridingPoliciesKeys := lo.FilterMap(lo.Uniq(lo.Flatten(lo.Values(overridingPolicies))), func(policyLocator string, _ int) (k8stypes.NamespacedName, bool) {
+			policyKey, err := kuadrantpolicymachinery.NamespacedNameFromLocator(policyLocator)
+			return policyKey, err == nil
+		})
+		return kuadrant.EnforcedCondition(policy, kuadrant.NewErrOverridden(policyKind, overridingPoliciesKeys), false)
+	}
+
+	var componentsToSync []string
+
+	// check the status of Limitador
+	if limitadorLimitsModified, stateLimitadorLimitsModifiedPresent := state.Load(StateLimitadorLimitsModified); stateLimitadorLimitsModifiedPresent && limitadorLimitsModified.(bool) {
+		componentsToSync = append(componentsToSync, kuadrantv1beta1.LimitadorGroupKind.Kind)
+	} else {
+		limitador := GetLimitadorFromTopology(topology)
+		if limitador == nil {
+			return kuadrant.EnforcedCondition(policy, kuadrant.NewErrSystemResource("limitador"), false)
+		}
+		if !meta.IsStatusConditionTrue(limitador.Status.Conditions, limitadorv1alpha1.StatusConditionReady) {
+			componentsToSync = append(componentsToSync, kuadrantv1beta1.LimitadorGroupKind.Kind)
+		}
+	}
+
+	// check the status of the gateways' configuration resources
+	for _, g := range affectedGateways {
+		controllerName := g.gatewayClass.Spec.ControllerName
+		switch defaultGatewayControllerName(controllerName) {
+		case defaultIstioGatewayControllerName:
+			// EnvoyFilter
+			istioTokenRateLimitClustersModifiedGateways, _ := state.Load(StateIstioTokenRateLimitClustersModified)
```

Comment:
Should we be doing the okay checks here?

Reply:
yeah, RLP does the same (copied again)